cmds/boot/netdiskboot/netdiskBoot.go

```diff
@@ -0,0 +1,234 @@
+package main
+
+import (
+	"bytes"
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/boot/kexec"
+	"github.com/u-root/u-root/pkg/mount"
+
+	"github.com/dustin/go-humanize"
+	"github.com/xi2/xz"
+)
+
+var (
+	// uroot.uinitargs='-debug' as kernel cmd args
+	doDebug      = flag.Bool("d", true, "Print debug output")
+	cfgfile      = "config.json"
+	cfg          *netdiskBootConfig
+	mountPath    = "/mnt/drive"
+	bootfilePath = "/mnt/drive/boot"
+)
+
+// WriteCounter counts the number of bytes written to it. It implements to the io.Writer interface
+// and we can pass this into io.TeeReader() which will report progress on each write cycle.
+type WriteCounter struct {
+	Total uint64
+}
+
+func (wc *WriteCounter) Write(p []byte) (int, error) {
+	n := len(p)
+	wc.Total += uint64(n)
+	wc.PrintProgress()
+	return n, nil
+}
+
+// PrintProgress prints progess of an ongoing process. In this case the downloading of a given image file.
+func (wc WriteCounter) PrintProgress() {
+	// Clear the line by using a character return to go back to the start and remove
+	// the remaining characters by filling it with spaces
+	fmt.Printf("\r%s", strings.Repeat(" ", 35))
+
+	// Return again and print current status of download
+	// We use the humanize package to print the bytes in a meaningful way (e.g. 10 MB)
+	fmt.Printf("\rDownloading... %s complete", humanize.Bytes(wc.Total))
+}
+
+var banner = `
+
+ _________________________________
+< Booting with 9e is even hotter! >
+ ---------------------------------
+                                /
+                     __       //
+                     -\= \=\ //
+                   --=_\=---//=--
+                 -_==/  \/ //\/--
+                  ==/   /O   O\==--
+     _ _ _ _     /_/    \  ]  /--
+    /\ ( (- \    /       ] ] ]==-
+   (\ _\_\_\-\__/     \  (,_,)--
+  (\_/                 \     \-
+  \/      /       (   ( \  ] /)
+  /      (         \   \_ \./ )
+  (       \         \      )  \
+  (       /\_ _ _ _ /---/ /\_  \
+   \     / \     / ____/ /   \  \
+    (   /   )   / /  /__ )   (  )
+    (  )   / __/ '---'       / /
+    \  /   \ \             _/ /
+    ] ]     )_\_         /__\/
+    /_\     ]___\
+   (___)
+`
+var debug = func(string, ...interface{}) {}
+
+func main() {
+	if *doDebug {
+		debug = log.Printf
+		log.Print("Debug mode!")
+	}
+
+	cfg, err := loadConfig(cfgfile)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+	log.Printf("\n%v", cfg.String())
+
+	log.Print(banner)
+
+	configureDHCPNetwork()
+
+	if !*doDebug {
```

Comment:
That whole part only is executed in debug mode? Maybe dry run?

Reply:
done